src/main/java/org/jenkinsci/plugins/pubsub/EventProps.java

```diff
@@ -116,6 +135,10 @@
          * Job run status/result.
          */
         job_run_status,
+        /**
+         * Job run SCM commit Id, if relevant.
+         */
+        job_run_commitId,
```

Comment:
@tfennelly will be good to know how does job_run_commitId get set. There has been bit of confusion w.r.t which action to pick to pick revision. in blueocean we are using `SCMRevisionAction` and then `SCMRevision.toString()` to get revision/commitId. Otherwise if picking `MultiScmRevisionState` or BuildData can give interim scm revision info and not what blueocean thinks it wants, that is of checkout scm repo's commit id.

Alternatively we can also use BlueMessageEnricher to compute commitId that we think is consistent. Here we get BlueRun from given jenkins model Run and get commitId.

```java
+        blueocean_job_run_commitId,
         blueocean_queue_item_expected_build_number
     }
 
@@ -72,7 +77,7 @@ public class BlueMessageEnricher extends MessageEnricher {
             }
             Link jobUrl = LinkResolver.resolveLink(jobChannelItem);
 
-            BlueOrganization org = OrganizationResolver.getInstance().getContainingOrg(jobChannelItem);
+            final BlueOrganization org = OrganizationResolver.getInstance().getContainingOrg(jobChannelItem);
             if (org!=null) {
                 message.set(EventProps.Jenkins.jenkins_org, org.getName());
             }
@@ -96,10 +101,25 @@ public class BlueMessageEnricher extends MessageEnricher {
                 if (blueQueueItem != null) {
                     jobChannelMessage.set(BlueEventProps.blueocean_queue_item_expected_build_number, Integer.toString(blueQueueItem.getExpectedBuildNumber()));
                 } else {
-                    Run run = QueueUtil.getRun(job, queueId);
+                    final Run run = QueueUtil.getRun(job, queueId);
                     if (run == null) {
                         return;
                     }
+                    BlueRun blueRun = null;
+                    for(BlueRunFactory blueRunFactory : BlueRunFactory.all()){
+                        blueRun = blueRunFactory.getRun(run, new Reachable() {
+                            @Override
+                            public Link getLink() {
+                                return org.getLink().rel("pipelines").rel(run.getParent().getName()).rel("runs");
+                            }
+                        });
+                        if(blueRun != null){
+                            break;
+                        }
+                    }
+                    if(blueRun != null && blueRun instanceof PipelineRunImpl){
+                        jobChannelMessage.set(BlueEventProps.blueocean_job_run_commitId, ((PipelineRunImpl) blueRun).getCommitId());
+                    }
```